components/react/hooks/src/useSwipe/index.js

```diff
@@ -0,0 +1,80 @@
+import {useCallback, useEffect, useState} from 'react'
+import useMount from '../useMount'
+
+const THRESHOLDS = {
+  DISTANCE: 100,
+  TIME: 100
+}
+
+export default function useSwipe(ref) {
+  const [start, setStart] = useState({x: 0, y: 0, time: 0})
+  const [capturing, setCapturing] = useState(false)
+  const [end, setEnd] = useState({x: 0, y: 0})
+  const [direction, setDirection] = useState(undefined)
+
+  const onStart = ({clientX: x, clientY: y}) => {
+    setCapturing(true)
+    setStart({x, y, time: +new Date()})
+  }
+
+  const onMove = useCallback(({clientX: x, clientY: y}) => setEnd({x, y}), [])
+
+  useEffect(() => {
+    const {current} = ref
+
+    if (current) {
+      if (capturing) {
+        current.addEventListener('pointermove', onMove)
+      } else {
+        current.removeEventListener('pointermove', onMove)
+
+        const {TIME} = THRESHOLDS
+        const deltaTime = +new Date() - start.time
```

Comment:
I think that `Date.now()` is faster than `+new Date()` and easier to understand 
```suggestion
        const deltaTime = Date.now() - start.time
```